Exception when running AWS client example code

Run from IntelliJ 

Using Http4s 0.23.26

```scala
addSbtPlugin("com.disneystreaming.smithy4s" % "smithy4s-sbt-codegen" % "0.18.12")
```

```scala
libraryDependencies ++= Seq(
	// version sourced from the plugin
	"com.disneystreaming.smithy4s" %% "smithy4s-aws-http4s" % smithy4sVersion.value,
	"com.disneystreaming.smithy" % "aws-dynamodb-spec" % "2023.09.22" % Smithy4s
)
```

```scala
object TestSmithy extends IOApp.Simple {

  def run: IO[Unit] = resource.use { dynamodb =>
    dynamodb
      .listTables(limit = Some(ListTablesInputLimit(10)))
      .flatMap(IO.println(_))
  }

  val resource: Resource[IO, DynamoDB[IO]] =
    for {
      httpClient <- EmberClientBuilder.default[IO].build
      awsEnv <- AwsEnvironment.default(httpClient, AwsRegion.US_EAST_1)
      dynamodb <- AwsClient(DynamoDB, awsEnv)
    } yield dynamodb

}
```

```
smithy4s.aws.AwsCredentialsFileException: Unable to parse credentials file.
	at smithy4s.aws.AwsCredentialsFileException$.apply(AwsCredentialsFile.scala:27)
	at smithy4s.aws.AwsCredentialsFile$.processFileLines$$anonfun$2(AwsCredentialsFile.scala:122)
	at cats.syntax.EitherOps$.leftMap$extension(either.scala:193)
	at smithy4s.aws.AwsCredentialsFile$.processFileLines(AwsCredentialsFile.scala:122)
	at smithy4s.aws.AwsCredentialsFile$.fromDisk$$anonfun$1(AwsCredentialsFile.scala:46)
	at map @ smithy4s.aws.AwsCredentialsFile$.fromDisk$$anonfun$1(AwsCredentialsFile.scala:48)
	at flatMap @ smithy4s.aws.AwsCredentialsFile$.fromDisk$$anonfun$1(AwsCredentialsFile.scala:69)
	at flatMap @ fs2.Compiler$Target.flatMap(Compiler.scala:163)
	at flatMap @ fs2.Compiler$Target.flatMap(Compiler.scala:163)
	at flatMap @ fs2.Compiler$Target.flatMap(Compiler.scala:163)
	at flatMap @ fs2.Compiler$Target.flatMap(Compiler.scala:163)
	at flatMap @ fs2.Compiler$Target.flatMap(Compiler.scala:163)
	at flatMap @ fs2.Compiler$Target.flatMap(Compiler.scala:163)
	at clear @ org.http4s.client.middleware.Retry$.retryLoop$1(Retry.scala:90)
	at flatMap @ fs2.Compiler$Target.flatMap(Compiler.scala:163)
	at rethrow$extension @ fs2.Compiler$Target.compile$$anonfun$1(Compiler.scala:157)
	at run$ @ com.comcast.cas.TestSmithy$.run(TestSmithy.scala:13)
	at get @ fs2.internal.Scope.openScope(Scope.scala:275)
	at flatMap @ fs2.Compiler$Target.flatMap(Compiler.scala:163)
	at flatMap @ fs2.Pull$.goCloseScope$1$$anonfun$1$$anonfun$3(Pull.scala:1217)
	at update @ fs2.internal.Scope.releaseChildScope(Scope.scala:227)
Caused by: scala.MatchError: [OneCloud/ApplicationUser@555170539925] (of class java.lang.String)
	at smithy4s.aws.AwsCredentialsFile$.lookingForProfile$1(AwsCredentialsFile.scala:105)
	at smithy4s.aws.AwsCredentialsFile$.processFileLines$$anonfun$1(AwsCredentialsFile.scala:118)
	at cats.syntax.EitherObjectOps$.catchNonFatal$extension(either.scala:391)
	at smithy4s.aws.AwsCredentialsFile$.processFileLines(AwsCredentialsFile.scala:119)
	at smithy4s.aws.AwsCredentialsFile$.fromDisk$$anonfun$1(AwsCredentialsFile.scala:46)
	at cats.effect.IOFiber.succeeded(IOFiber.scala:1204)
	at cats.effect.IOFiber.runLoop(IOFiber.scala:245)
	at cats.effect.IOFiber.asyncContinueSuccessfulR(IOFiber.scala:1372)
	at cats.effect.IOFiber.run(IOFiber.scala:113)
	at cats.effect.unsafe.WorkerThread.run(WorkerThread.scala:743)
```

Redacted credentials:

[OneCloud/ApplicationUser@USER_ID]
aws_access_key_id     = KEY_ID
aws_secret_access_key = ACCESS_KEY
aws_session_token     = a//////////b+c/d+e/f/g/h+i/j+k/l/m//n/o+p/q

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,7 @@
 # 0.18.15
 
 * Add support for injecting endpoint-specific middlewares in AWS clients
+* Fixes a bug in the parsing of AWS credentials files
 
 # 0.18.14
 

--- a/modules/aws-http4s/src/smithy4s/aws/AwsCredentialsFile.scala
+++ b/modules/aws-http4s/src/smithy4s/aws/AwsCredentialsFile.scala
@@ -168,7 +168,7 @@ object AwsCredentialsFile {
 }
 
 private object Profile {
-  private val profileMatch = "([\\w_-]*)"
+  private val profileMatch = "(.*)"
   object Default {
     def unapply(s: String): Option[String] =
       if (s == "[default]") Some("default") else None
@@ -177,7 +177,7 @@ private object Profile {
   object Prefixed {
     private val reg = s"^\\[profile $profileMatch\\]$$".r
     def unapply(s: String): Option[String] = s match {
-      case reg(first) => Some(first)
+      case reg(first) => Some(first.trim())
       case _          => None
     }
   }
@@ -185,7 +185,7 @@ private object Profile {
   object NonPrefixed {
     private val reg = s"^\\[$profileMatch\\]$$".r
     def unapply(s: String): Option[String] = s match {
-      case reg(first) => Some(first)
+      case reg(first) => Some(first.trim())
       case _          => None
     }
   }

--- a/modules/aws-http4s/test/src/smithy4s/aws/AwsCredentialsFileTest.scala
+++ b/modules/aws-http4s/test/src/smithy4s/aws/AwsCredentialsFileTest.scala
@@ -26,7 +26,10 @@ object AwsCredentialsFileTest extends FunSuite {
       e: Either[Throwable, A]
   )(f: A => Expectations): Expectations =
     e.fold(
-      err => failure(s"Got Left but expected a Right. ${err.getMessage}"),
+      err => {
+        err.printStackTrace()
+        failure(s"Got Left but expected a Right. ${err.getMessage}")
+      },
       f
     )
 
@@ -141,6 +144,21 @@ object AwsCredentialsFileTest extends FunSuite {
       expect.same(Some(creds), res.default) &&
       expect.same(creds, res.profiles("p1")) &&
       expect.same(creds, res.profiles("p2"))
+    }
+  }
+
+  test("profiles can have special characters in them") {
+    expectRight(
+      AwsCredentialsFile.processFileLines(
+        asLines(
+          """|[Namespace/User@USER_ID]
+             |aws_access_key_id = key
+             |aws_secret_access_key = sec
+             |aws_session_token = token""".stripMargin
+        )
+      )
+    ) { res =>
+      expect.same(creds, res.profiles("Namespace/User@USER_ID"))
     }
   }
 

